### ISSUE TITLE
Newman 6.1.0 seems to have a memory leak: crash after enough iterations

### ISSUE THREAD

**Issue by mallechilie** [a85/POSTMan-Chrome-Extension#12579]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

I have a collection of requests that I run for many iterations (1000+). For the old version (6.0.0) of newman, this is runs fine, and newman takes about 150-500mb of memory. This fluctuates a bit over time, but doesn't accumulate. 

When doing the exact same thing in newman 6.1.0, the memory reaches 2gb by the 10th iteration. When the memory runs out, the following error is shown:
![afbeelding](https://github.com/postmanlabs/postman-app-support/assets/16206374/c9e78007-59b9-4493-bf22-cb6d819b865d)

I don't have a large collection that I can sharefor this sadly.

### Steps To Reproduce

1. run `npm install -g newman@6.0.0`
2. run a large collection a high number of iterations
3. it runs fine.

---
1. run `npm install -g newman@6.1.0`
2. run a large collection a high number of iterations
3. it crashes.

### Operating System

Windows

### Postman Platform

Newman

### Newman Version

6.1.0

### User Account Type

Signed In User

**Comment by Jaiaggarwaaaaal** [comment]
Try to use npm install -g newman 
You can follow the commands mention in this link https://writewithjai.medium.com/day-18-8325997aa3c1

**Comment by mallechilie** [comment]
@Jaiaggarwaaaaal I have installed newman as you can see in the issue. 

I run my requests using the following code, which runs fine in both versions. (Except in 6.1.0 it shows a memory leak).

```batch
SET postman_collection=".\Ticket export.postman_collection.json"
SET postman_environment=.\halopsa.postman_environment.json
SET iterations=1000


for /l %%x in (1, 1, 20) do (
	newman run %postman_collection% -e %postman_environment% -n %iterations% --export-globals .\export_global_variables_%%x.json
)
```

Can you explain to me how running `npm install -g newman` would help resolve the memory issue?

**Comment by vedkribhu** [comment]
@mallechilie I am able to replicate the issue on my end. We are working on a fix for this.

**Comment by mallechilie** [comment]
@vedkribhu Awesome, thank you so much!

https://github.com/postmanlabs/postman-app-support/issues/12570 This issue may be related

**Comment by vedkribhu** [comment]
@mallechilie we have addressed the issue in v6.1.1. Let me know if you are still facing it.
